// LINT: ALLOW_GROUPS
// Protocol buffer specifications for task configuration.

syntax = "proto2";

package syntaxnet;

// Task input descriptor.
message TaskInput {
  // Name of input resource.
  required string name = 1;

  // Name of stage responsible of creating this resource.
  optional string creator = 2;

  // File format for resource.
  repeated string file_format = 3;

  // Record format for resource.
  repeated string record_format = 4;

  // Is this resource multi-file?
  optional bool multi_file = 5 [default = false];

  // An input can consist of multiple file sets.
  repeated group Part = 6 {
    // File pattern for file set.
    optional string file_pattern = 7;

    // File format for file set.
    optional string file_format = 8;

    // Record format for file set.
    optional string record_format = 9;
  }
}

// A task specification is used for describing executing parameters.
message TaskSpec {
  // Task parameters.
  repeated group Parameter = 3 {
    required string name = 4;
    optional string value = 5;
  }

  // Task inputs.
  repeated TaskInput input = 6;

  reserved 1, 2, 7;
  reserved "task_name", "task_type", "output";
}
